src/Textual/transform.jl

```diff
@@ -0,0 +1,22 @@
+"""
+    replace_all_caps(String)
+
+Replace tokens in ALL CAPS by their lower version and add xxup before.
+"""
+
+function replace_all_caps(t)
+    t = replace(t, r"([A-Z]+[^a-z\s]*)(?=(\s|$))" => s"xxup \1")
+    return replace(t, r"([A-Z]*[^a-z\s]+)(?=(\s|$))" => lowercase)
+end
+
+"""
```

Comment:
Some inline tests for these preprocessing steps would be good to have, too

Reply:
Yeah, that's in my to-do list. Just wanted to make sure they're necessary. I'll add some tests in my next commit.